gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -168,4 +170,32 @@ public static Writable deserializeFromString(Class<? extends Writable> writableC
       closer.close();
     }
   }
+
+  /**
+   * Serializes a given {@link FsPermission} object into a {@link String} and stores it as a key, value pair in the
+   * given {@link State}. This method is meant to be used in conjunction with
+   * {@link #deserializeFsPermission(State, String)}.
+   */
+  public static void serializeFsPermission(State state, String key, FsPermission fsPermissions) {
+    state.setProp(key, String.format("%04o", fsPermissions.toShort()));
+  }
+
+  /**
+   * Serializes a given {@link String} that represents the permissions of a file / directory in octal notation. The
+   * {@link String} is stored in the given {@link State}. This method is meant to be used in conjunction with
+   * {@link #deserializeFsPermission(State, String)}.
+   */
+  public static void serializeOctalPermissions(State state, String key, String octalPermissions) {
+    state.setProp(key, octalPermissions);
+  }
+
+  /**
+   * Deserializes a {@link FsPermission} object from the given {@link State}. This method is meant to be used in
+   * conjunction with {@link #serializeFsPermission(State, String, FsPermission)} or
+   * {@link #serializeOctalPermissions(State, String, String)}.
+   */
+  public static FsPermission deserializeFsPermission(State state, String key) {
+    return new FsPermission(state.getPropAsShortWithRadix(key,
+        FsPermission.getDefault().toShort(), ConfigurationKeys.PERMISSION_PARSING_RADIX));
```

Comment:
Thats fine though right, since the short is converted in the `String.format` code block. So radix = 8 should handle setting the short and setting in octal.